From 99978834faa678f6d669044613e978f4a21910b9 Mon Sep 17 00:00:00 2001
From: OpenWrt community <openwrt-devel@lists.openwrt.org>
Date: Thu, 16 Mar 2023 17:03:56 +0530
Subject: [PATCH 051/500] OpenWrt: 500-fs_cdrom_dependencies.patch

fs: add cdrom dependency

Change-Id: I8eaefccbed444012a8a6fb1e168f31923b6685be
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 058b955d0d4201a9efb8a8d9e641f15bc8da92f3)
---
 fs/hfs/Kconfig     | 1 +
 fs/hfsplus/Kconfig | 1 +
 fs/isofs/Kconfig   | 1 +
 fs/udf/Kconfig     | 1 +
 4 files changed, 4 insertions(+)

diff --git a/fs/hfs/Kconfig b/fs/hfs/Kconfig
index 5ea5cd8ecea9..8a0ca259dc58 100644
--- a/fs/hfs/Kconfig
+++ b/fs/hfs/Kconfig
@@ -2,6 +2,7 @@
 config HFS_FS
 	tristate "Apple Macintosh file system support"
 	depends on BLOCK
+	select CDROM
 	select BUFFER_HEAD
 	select NLS
 	select LEGACY_DIRECT_IO
diff --git a/fs/hfsplus/Kconfig b/fs/hfsplus/Kconfig
index 8ce4a33a9ac7..f3f07878dbbd 100644
--- a/fs/hfsplus/Kconfig
+++ b/fs/hfsplus/Kconfig
@@ -2,6 +2,7 @@
 config HFSPLUS_FS
 	tristate "Apple Extended HFS file system support"
 	depends on BLOCK
+	select CDROM
 	select BUFFER_HEAD
 	select NLS
 	select NLS_UTF8
diff --git a/fs/isofs/Kconfig b/fs/isofs/Kconfig
index 51434f2a471b..9847d2e3aac2 100644
--- a/fs/isofs/Kconfig
+++ b/fs/isofs/Kconfig
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config ISO9660_FS
 	tristate "ISO 9660 CDROM file system support"
+	select CDROM
 	select BUFFER_HEAD
 	help
 	  This is the standard file system used on CD-ROMs.  It was previously
diff --git a/fs/udf/Kconfig b/fs/udf/Kconfig
index 8f7ce30d47fd..0b015b1a455a 100644
--- a/fs/udf/Kconfig
+++ b/fs/udf/Kconfig
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config UDF_FS
 	tristate "UDF file system support"
+	select CDROM
 	select BUFFER_HEAD
 	select CRC_ITU_T
 	select NLS
-- 
2.34.1

